x-pack/plugins/monitoring/server/lib/logstash/get_pipeline.ts

```diff
@@ -76,6 +76,17 @@ export function _enrichStateWithStatsAggregation(
   statsAggregation: any,
   timeseriesIntervalInSeconds: number
 ) {
+  // we could have data in both legacy and metricbeat collection, we pick the bucket most filled
```

Comment:
Wonder if we should consider plotting both? I'd guess this will mean a break in the graph when moving from internal collection to metricbeat.

Since the pipeline stack monitoring is beta though, maybe that's okay.

Reply:
That could be an option. We could also merge the aggregations and ensure we only consider a single datasource per node, since a pipeline can run on multiple nodes if we receive both legacy and metricbeat monitoring for a pipeline-node tuple there is duplication that we should ignore

Reply:
That's a really good point. I opened https://github.com/elastic/kibana/issues/123191 to capture the improvement for later.